xlators/features/snapview-client/src/snapview-client.c

```diff
@@ -2149,8 +2149,10 @@ gf_svc_readdirp(call_frame_t *frame, xlator_t *this, fd_t *fd, size_t size,
     wind = _gf_true;
 
 out:
-    if (!wind)
+    if (!wind) {
         SVC_STACK_UNWIND(readdirp, frame, op_ret, op_errno, &entries, NULL);
+        svc_local_free(local);
```

Comment:
Won't SVC_STACK_UNWIND do this if frame->local is not null?

Reply:
It is better to do assign local to frame-local immediately after a successful allocation. Then we don't need to free it explicitly. Not sure if other FOPS in this file have the same issue.

Reply:
Yes we could assign it to frame->local immediately.
Coverity is not complaining resource leak in any other fops in this file. So they should not have this issue.

Reply:
Done.